src/scripts/main.js

```diff
@@ -1,12 +1,39 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handler = (e) => {
+      // Remove the event listener after it is triggered
+      element.removeEventListener(eventName, handler);
+
+      // Resolve the promise with the message
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    };
+
+    // Attach the event listener to the element
+    element.addEventListener(eventName, handler);
+  });
+}
+
+// Function to print the message to the DOM
+function printMessage(message) {
+  const messageDiv = document.createElement('div');
+
+  messageDiv.classList.add('message');
+  messageDiv.textContent = message;
+  document.body.appendChild(messageDiv);
 }
 
-const printMessage = (message) => {
-  // write your code here
-};
+// Example usage: Add event listeners on input and button elements
+document.querySelectorAll('input, button').forEach((element) => {
+  ['click', 'input', 'blur'].forEach((e) => {
+    waitFor(element, e).then((message) => {
+      printMessage(message); // Call printMessage to add the message to the DOM
+    });
+  });
+});
```

Comment:
The loop on lines 30-36 is adding multiple event listeners to each element for the same events ('click', 'input', 'blur'). This can lead to multiple messages being printed for the same event. Consider checking if the event listener already exists before adding it, or refactor the logic to prevent redundant listeners.